_docs/implementation.md

```diff
@@ -0,0 +1,104 @@
+# Introduction
+This proposal outlines a method for injecting runtime hooks into target functions in Go programs, enabling dynamic monitoring and modification of function behavior. The approach leverages trampoline code injection and function pointer redirection to seamlessly integrate monitoring logic without requiring significant changes to the user's codebase.
+
+The goal is to provide a flexible and non-intrusive mechanism for instrumenting Go applications, particularly for use cases such as observability (e.g., OpenTelemetry), debugging, or performance profiling.
+
+# Core Principles
+## 1. Trampoline Code Injection
+We inject trampoline code into the Target (lib-side) function, which ultimately jumps to the actual monitoring code via the function pointer Hook.
```

Comment:
To be precise, it's trampoline function call
```suggestion
We inject trampoline function call into the Target (lib-side) function, which ultimately jumps to the actual monitoring code via the function pointer Hook.
```

Reply:
Just to avoid additional questions and confusions, I think it would be good to explain why are we doing two-level indirection, first `Trampoline` call, then hook, instead of let's say directly injecting opentelemetry code.